### ISSUE TITLE
Review comment for a pull request with subject_type

### ISSUE THREAD

**Issue by vanya20074** [PyGithub/PyGithub#2640]
Hi,
I am trying to create comment with `subject_type:FILE` but got an error:
```
{
    "message": "Invalid request.\n\nNo subschema in \"oneOf\" matched.\n\"position\" wasn't supplied.\n\"in_reply_to\" wasn't supplied.\n\"subject_type\" is not a permitted key.\n\"line\" wasn't supplied.\nFILE is not a member of [\"line\", \"file\"].",
    "documentation_url": "https://docs.github.com/rest/pulls/comments#create-a-review-comment-for-a-pull-request"
}
```
At the same time `subject_type:file` working fine but [create_review_comment](https://github.com/PyGithub/PyGithub/blob/main/github/PullRequest.py#L450) have check with uppercase:

```
        assert subject_type is github.GithubObject.NotSet or subject_type in [
            "LINE",
            "FILE",
            "side",
        ], subject_type
```
Am I doing something wrong?

**Comment by EnricoMi** [comment]
The docs say it's uppercase, so the assertion looks right: https://docs.github.com/en/rest/pulls/comments?apiVersion=2022-11-28#create-a-review-comment-for-a-pull-request

Please check with Github Support whether upper or lowercase is expected. You could also try to execute both calls via curl to know for sure what is expected by the Github API. Please report back with your findings.

**Comment by EnricoMi** [comment]
Btw., #2641.

**Comment by vanya20074** [comment]
I have tested calls via curl with upper and lowercase.
With lower is ok - 201 created
With uppercase - 422Unprocessable Entity
Also asked in github for [clarification](https://github.com/github/docs/issues/26844), but I think it's only mistake in documentation

**Comment by EnricoMi** [comment]
I have commented on that issue, waiting for confirmation, but I think we have to fix those assertions then.

Would you mind open a PR?

**Comment by vanya20074** [comment]
Yes, sure, please take a look https://github.com/PyGithub/PyGithub/pull/2642

**Comment by EnricoMi** [comment]
Fixed via #2642
